### ISSUE TITLE
Is it possible to train and detect lidar point cloud data using yolov8? 

### ISSUE THREAD

**Issue by safal-305** [ultralytics/ultralytics#1058]
### Question

This is not an issue , but more like a doubt or clarification. I was wondering if it as possible to detect yolov8 for the detection and training of objects from lidar point cloud data.

**Comment by JustasBart** [comment]
I've an **Intel® RealSense™ LiDAR Camera L515** myself and would love to know also :rocket:

**Comment by AyushExel** [comment]
Not currently as we don't have the right data loader and loss functions(I'm not sure what exactly it'll be) in place. But I think it'll be a nice experiment if you decide to embark on that journey. Happy to help!

**Comment by nguyenlam185** [comment]
@JustasBart 
Check this out https://github.com/ghimiredhikura/Complex-YOLOv3

**Comment by safal-305** [comment]
> @JustasBart Check this out https://github.com/ghimiredhikura/Complex-YOLOv3

did you try this repo?

**Comment by JustasBart** [comment]
@safal-305 @nguyenlam185 Will check it out in the next few days...

**Comment by safal-305** [comment]
@nguyenlam185 @JustasBart I ran the test-detection sampledata, but it retuned nothing.

**Comment by Prashambhuta** [comment]
I was able to run test, and evaluation for this repo: https://github.com/maudzung/Complex-YOLOv4-Pytorch. I am looking to train on Yolo8, and see improvements in performance. Has somebody tried running training on this model?

**Comment by safal-305** [comment]
> I was able to run test, and evaluation for this repo: https://github.com/maudzung/Complex-YOLOv4-Pytorch. I am looking to train on Yolo8, and see improvements in performance. Has somebody tried running training on this model?

@Prashambhuta Did you do the test on your own lidar data using the preloaded trained model? or just on the given velodyne data itself?

**Comment by Prashambhuta** [comment]
@safal-305 on the given Kitti Velodyne data.

**Comment by Prashambhuta** [comment]
I am trying to replicate this training, but replacing the YOLO architecture with YOLOv8. 

https://github.com/AI-liu/Complex-YOLO/blob/master/main.py

**Comment by glenn-jocher** [comment]
@Prashambhuta you can start by modifying the model architecture and the loss function to match that of YOLOv8, while keeping the rest of the code intact. You may also need to adjust some of the hyperparameters, depending on the specifics of the dataset and the YOLOv8 model configuration.

Keep in mind that YOLOv8 is not an official YOLO variant recognized by the original YOLO creators, so you may run into additional challenges that are specific to this implementation.

Overall, it should be possible to adapt the code to work with YOLOv8. If you encounter any issues or have additional questions, feel free to reach out for further assistance.

**Comment by Petros626** [comment]
@Prashambhuta can I ask you a questions to train yolov8 obb with lidar data. It's more about data format, loading etc.

**Comment by Petros626** [comment]
Finally solved it using an official kitti evaluation code (https://github.com/open-mmlab/OpenPCDet/tree/master/pcdet/datasets/kitti/kitti_object_eval_python, https://github.com/traveller59/kitti-object-eval-python):

![Image](https://github.com/user-attachments/assets/1ef78942-f433-4d10-8349-56a68798d66e)

You can check my repos, which I used to implement a custom kitti bev dataset, training the yolo detector and evaluate it.

**Comment by glenn-jocher** [comment]
@Petros626 excellent work implementing YOLOv8 OBB with lidar data! Your custom KITTI BEV dataset approach and successful results will be valuable for others looking to work with point cloud data. Thanks for sharing your implementation and making it available to the community.
